app/controllers/jobs_controller.rb

```diff
@@ -20,25 +38,47 @@ def create
     end
   end
 
-  def show
+  def update
+    @company = Company.find(params[:company_id])
     @job = Job.find(params[:id])
+    if @job.update(job_params)
+      flash[:success] = "#{@job.title} updated!"
+      redirect_to company_job_path(@company, @job)
+    else
+      @categories = Category.all
+      render :edit
+    end
   end
 
-  def edit
-    # implement on your own!
+  def destroy
+    job = Job.find(params[:id])
+    job.destroy
+
+    flash[:success] = "#{job.title} was successfully deleted!"
+    redirect_to company_jobs_path
   end
 
-  def update
-    # implement on your own!
+  def sort
+    if params[:location]
+      @jobs = Job.where(city: params[:location])
+      @message = "All jobs in #{params[:location]}"
+    elsif params[:sort] == "interest"
+      @jobs = Job.order(:level_of_interest)
+      @message = "Jobs sorted by level of interest"
+    else
+      @jobs = Job.all
+      @message = "All Jobs"
+    end
   end
 
-  def destroy
-    # implement on your own!
+  def dashboard
```

Comment:
to follow REST, we might want to use a dashboard controller which would have an index and show or a QueryController for the other stuff